src/Hosting/WindowsServices/src/WindowsServiceLifetime.cs

```diff
@@ -72,6 +78,7 @@ protected override void OnStart(string[] args)
         protected override void OnStop()
         {
             ApplicationLifetime.StopApplication();
+            _delayStop.Wait();
```

Comment:
This should either time out or you should add IDisposible to this class and also call _delayStop.Set() there. There are error conditions where ApplicationStopped might not be called but Dispose would be.

Reply:
I ended up doing both to be safe